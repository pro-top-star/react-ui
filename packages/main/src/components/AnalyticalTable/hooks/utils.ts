export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
