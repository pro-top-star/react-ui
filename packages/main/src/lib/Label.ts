import { Label } from '../webComponents/Label';

export { Label };
