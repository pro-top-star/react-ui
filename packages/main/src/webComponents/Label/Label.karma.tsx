import { mountThemedComponent } from '@shared/tests/utils';
import { expect, use } from 'chai';
import { matchSnapshot } from 'chai-karma-snapshot';
import React from 'react';
import { Label } from '../../lib/Label';

use(matchSnapshot);

describe('Label', () => {
  it('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Label />);
    expect(wrapper.debug()).to.matchSnapshot();
  });
});
