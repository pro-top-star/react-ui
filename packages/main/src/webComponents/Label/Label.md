[Link to UI5 Web Components Playground](https://sap.github.io/ui5-webcomponents/playground/components/Label)

```jsx
import { Label } from '@ui5/webcomponents-react/lib/Label';
```
