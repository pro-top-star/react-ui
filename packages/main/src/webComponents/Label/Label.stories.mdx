import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs/blocks';
import { Label } from '@ui5/webcomponents-react/lib/Label';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="UI5 Web Components / Label"
  component={Label}
  argTypes={{
    ...createSelectArgTypes({}),
    children: { control: 'text' },
    ...DocsCommonProps
  }}
  args={{
    children: 'Label Text',
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Label {...args} />;
    }}
  </Story>
</Canvas>

<ArgsTable story="." />
