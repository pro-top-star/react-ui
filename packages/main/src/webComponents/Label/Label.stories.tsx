import { boolean, text } from '@storybook/addon-knobs';
import { Label } from '@ui5/webcomponents-react/lib/Label';
import React from 'react';

export default {
  title: 'UI5 Web Components / Label',
  component: Label
};

export const generatedDefaultStory = () => (
  <Label
    for={text('for', '')}
    required={boolean('required', false)}
    showColon={boolean('showColon', false)}
    wrap={boolean('wrap', false)}
  >
    Some Content
  </Label>
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
