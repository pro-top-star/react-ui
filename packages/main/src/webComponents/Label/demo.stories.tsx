import { boolean, text } from '@storybook/addon-knobs';
import { Label } from '@ui5/webcomponents-react/lib/Label';
import React from 'react';

export default {
  title: 'UI5 Web Components / Label',
  component: Label
};

export const story = () => (
  <Label required={boolean('required', false)} wrap={boolean('wrap', false)} for={text('for', 'generatedString')}>
    {text('children', 'Some Content')}
  </Label>
);
story.story = {
  name: 'Default Story'
};
