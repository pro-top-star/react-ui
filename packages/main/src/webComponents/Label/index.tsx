import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Label from '@ui5/webcomponents/dist/Label';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface LabelPropTypes extends WithWebComponentPropTypes {
  required?: boolean; // @generated
  wrap?: boolean; // @generated
  for?: string; // @generated
  children?: string; // @generated
}

/**
 * <code>import { Label } from '@ui5/webcomponents-react/lib/Label';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/Label" target="_blank">UI5 Web Components Playground</a>
 */
const Label: FC<LabelPropTypes> = withWebComponent<LabelPropTypes>(UI5Label);

Label.displayName = 'Label';

Label.defaultProps = {
  for: '' // @generated
};

export { Label };
