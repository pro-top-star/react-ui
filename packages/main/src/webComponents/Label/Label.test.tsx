import { mountThemedComponent } from '@shared/tests/utils';
import { Label } from '@ui5/webcomponents-react/lib/Label';
import React from 'react';

describe('Label', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Label />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
