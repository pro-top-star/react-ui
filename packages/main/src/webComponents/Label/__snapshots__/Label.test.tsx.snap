// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Basic Test (generated) 1`] = `
<ui5-label
  for=""
/>
`;
